src/controller.js

```diff
@@ -0,0 +1,136 @@
+(function exportcontroller() {
+  function Controller(ship) {
+    this.ship = ship;
+    this.sailInterval = null; 
+    this.initForm();
+    document.querySelector('#sailbutton').addEventListener('click', () => {
+      this.setSail();
+     
+    });
+    this.initialiseSea();
+
+  }
+
+  Controller.prototype.initialiseSea = function initialiseSea() {
+    const backgrounds = [
+      './images_ship/water0.png',
+      './images_ship/water1.png',
+    ];
+    let backgroundIndex = 0;
+    const renderPorts = () => {
+      const portsElement = document.querySelector('#ports');
+      portsElement.style.width = '0px';
+      portsElement.innerHTML = ''; 
+
+      this.ship.itinerary.ports.forEach((port, index) => {
+        const newPortElement = document.createElement('div');
+        newPortElement.className = 'port';
+        newPortElement.dataset.portName = port.name;
+        newPortElement.dataset.portIndex = index;
+
+        portsElement.appendChild(newPortElement);
+        const portsElementWidth = parseInt(portsElement.style.width, 10);
+        portsElement.style.width = `${portsElementWidth + 256}px`;
+      });
+    };
+
+    const renderShip = () => {
+      const ship = this.ship;
+      const currentPortIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+      const nextPortIndex = currentPortIndex;
+      const nextPortElement = document.querySelector(
+        `[data-port-index='${nextPortIndex}']`
+      );
+
+      const shipElement = document.querySelector('#ship');
+      shipElement.style.top = `${nextPortElement.offsetTop + 32}px`;
+      shipElement.style.left = `${nextPortElement.offsetLeft - 32}px`;
+    };
+
+    renderPorts();
+    renderShip();
```

Comment:
I would rethink how you've got this logic setup. this function should just be solely focused on `initialiseSea`. Your other logic should be taken out of this function and put into their own functions, seperate, functions.